File: examples/simple-config/src/main/java/com/riversoforion/lena/example/SimpleConfigExample.java
/*
 * Copyright (c) 2024. Eric McIntyre / Rivers of Orion
 */
package com.riversoforion.lena.example;

import java.io.PrintStream;

public class SimpleConfigExample extends ExampleApplication {

    private final ApplicationConfig config = new ApplicationConfig();

    public static void main(String[] args) {

        System.out.println("=== Simple Configuration Example ===");
        SimpleConfigExample example = new SimpleConfigExample();

        if (example.isHelpRequested(args)) {
            example.printHelp(System.out);
            return;
        }

        example.printConfig(System.out);
    }

    public void printConfig(PrintStream out) {

        out.printf("%25s = %s%n", "serviceUrl", config.serviceUrl());
        out.printf("%25s = %s%n", "serviceApiKey", config.serviceApiKey());
        out.printf("%25s = %s%n", "serviceApiSecret", config.serviceApiSecret());
        out.printf("%25s = %d%n", "netConnectionTimeout", config.netConnectionTimeout());
        out.printf("%25s = %d%n", "netReadTimeout", config.netReadTimeout());
        out.printf("%25s = %b%n", "isLocalMode", config.isLocalMode());
    }

    @Override
    protected void printHelp(PrintStream out) {

        out.print("""
                  Demonstrates a simple usage of ConfigurationProperties
                  
                  Run with environment variables or system properties set. The following configuration properties are supported:
                      Environment Variable    System Property         Type
                      SERVICE_URL             service.url             string
                      SERVICE_API_KEY         service.api.key         string
                      SERVICE_API_SECRET      service.api.secret      string
                      NET_CONNECTION_TIMEOUT  net.connection.timeout  long
                      NET_READ_TIMEOUT        net.read.timeout        long
                      LOCAL_MODE              local.mode              flag
                  """);
    }
}

File: examples/simple-config-kotlin/src/commonMain/kotlin/com/riversoforion/lena/example/SimpleConfigExample.kt
/*
 * Copyright (c) 2024. Eric McIntyre / Rivers of Orion
 */
package com.riversoforion.lena.example

import org.riversoforion.lena.config.Namespace

class SimpleConfigExample {

    private val config: ApplicationConfig = createApplicationConfig(
        defaultConfigurationSource(),
        Namespace.root()
    )

    fun run(args: Array<String>) {
        println("=== Simple Configuration Example (Kotlin) ===")

        if (args.contains("-h") || args.contains("--help")) {
            printHelp()
            return
        }

        printConfig()
    }

    private fun printConfig() {
        println("${"serviceUrl".padStart(25)} = ${config.serviceUrl()}")
        println("${"serviceApiKey".padStart(25)} = ${config.serviceApiKey()}")
        println("${"serviceApiSecret".padStart(25)} = ${config.serviceApiSecret()}")
        println("${"netConnectionTimeout".padStart(25)} = ${config.netConnectionTimeout()}")
        println("${"netReadTimeout".padStart(25)} = ${config.netReadTimeout()}")
        println("${"isLocalMode".padStart(25)} = ${config.isLocalMode()}")
    }

    private fun printHelp() {
        println("""
            Demonstrates a simple usage of ConfigurationProperties (Kotlin/KMP)
            
            Run with environment variables or system properties set. The following configuration properties are supported:
                Environment Variable    System Property         Type
                SERVICE_URL             service.url             string
                SERVICE_API_KEY         service.api.key         string
                SERVICE_API_SECRET      service.api.secret      string
                NET_CONNECTION_TIMEOUT  net.connection.timeout  long
                NET_READ_TIMEOUT        net.read.timeout        long
                LOCAL_MODE              local.mode              flag
        """.trimIndent())
    }
}

fun main(args: Array<String>) {
    SimpleConfigExample().run(args)
}
